pvlib/iotools/pvgis.py

```diff
@@ -0,0 +1,175 @@
+"""
+Get, read, and parse data from `PVGIS <https://ec.europa.eu/jrc/en/pvgis>`_.
+
+For more information, see the following links:
+* `Interactive Tools <https://re.jrc.ec.europa.eu/pvg_tools/en/tools.html>`_
+* `Data downloads <https://ec.europa.eu/jrc/en/PVGIS/downloads/data>`_
+* `User manual docs <https://ec.europa.eu/jrc/en/PVGIS/docs/usermanual>`_
+
+More detailed information about the API for TMY and hourly radiation are here:
+* `TMY <https://ec.europa.eu/jrc/en/PVGIS/tools/tmy>`_
+* `hourly radiation
+  <https://ec.europa.eu/jrc/en/PVGIS/tools/hourly-radiation>`_
+* `daily radiation <https://ec.europa.eu/jrc/en/PVGIS/tools/daily-radiation>`_
+* `monthly radiation
+  <https://ec.europa.eu/jrc/en/PVGIS/tools/monthly-radiation>`_
+"""
+import io
+import requests
+import pandas as pd
+from pvlib.iotools import parse_epw
+
+URL = 'https://re.jrc.ec.europa.eu/api/'
+
+
+def get_pvgis_tmy(lat, lon, outputformat='json', usehorizon=True,
+                  userhorizon=None, startyear=None, endyear=None, url=URL,
+                  timeout=30):
+    """
+    Get TMY data from PVGIS. For more information see documentation for PVGIS
+    `TMY tools <https://ec.europa.eu/jrc/en/PVGIS/tools/tmy>`_
+
+    Parameters
+    ----------
+    lat : float
+        Latitude in degrees north
+    lon : float
+        Longitude in dgrees east
+    outputformat : string [default 'json']
+        Must be in ``['csv', 'basic', 'epw', 'json']``. See link for more info.
+    usehorizon : bool [default True]
+        include effects of horizon
+    userhorizon : list of float [default None]
+        elevation of horizon in degrees, at equally spaced azimuth clockwise
+        from north, _EG_: given 8 horizon elevations, the corresponding azimuth
+        are north, north-east, east, south-east, south, south-west, west, and
+        north-west
+    startyear : integer [default None]
+        first year to calculate TMY
+    endyear : integer [default None]
+        last year to calculate TMY, must be at least 10 years from first year
+    url : str [default :const:`~pvlib.iotools.pvgis.URL`]
+        base url of PVGIS API, append ``tmy`` to get TMY endpoint
+    timeout : int, default 30
+        time in seconds to wait for server response before timeout
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        the weather data
+    months_selected : list
+        TMY year for each month, ``None`` for basic and EPW
+    inputs : dict
+        the inputs, ``None`` for basic and EPW
+    meta : list or dict
+        meta data, ``None`` for basic
```

Comment:
Not without using `groupby`. Since it's already there and some people may ask for it or expect this to work the same as the existing API, I'm fine with keeping it, even if it's not that useful.